src/vaillant_netatmo_api/thermostat.py

```diff
@@ -93,6 +95,65 @@ async def async_get_thermostats_data(self) -> list[Device]:
 
         return [Device(**device) for device in body["body"]["devices"]]
 
+    async def async_get_dhw_data(
+        self,
+        device_id: str,
+        home_id: str,
+    ) -> None:
+        """
+        Get DHW data from the Netatmo API.
+
+        """
+
+        path = _GET_DHW_DATA_PATH
+        data = {
+            "device_id": device_id,
+            "home_id": home_id,
+        }
+
+        body = await self._post(
+            path,
+            data=data,
+        )
+
+        if body["status"] != _RESPONSE_STATUS_OK:
+            raise NonOkResponseException("Unknown response error. Check the log for more details.", path=path, data=data, body=body)
+
+        return body["body"]["home"]["modules"][0]["dhw_always_on"]
+
+    async def async_set_dhw_data(
+        self,
+        device_id: str,
+        home_id: str,
+        dhw_always_on: bool,
+    ) -> None:
+        """
+        Set DHW data as json post to the Netatmo API.
+
+        """
+
+        path = _SET_DHW_DATA_PATH
+        data = {
+            "home": {
+                "modules":[
+                    {
+                        "dhw_always_on":dhw_always_on,
+                        "id":device_id
+                    }
```

Comment:
What happens if you send `dhw_setpoint_temperature` attribute in this call? If it changes, then this API is not "change dhw always on", its "change `ConfigModule`", which means it should resemble the signature of `async_sync_schedule` method:

```python
    async def async_set_dhw_data(
        self,
        home_id: str,
        config_modules: list[ConfigModule],
    ) -> None:
```

Or if we don't want to use the same `ConfigModule` _read model_ because _write model_ is different (eg. doesn't have `rf_stats`, `dhw_temperature_min`, or `dhw_temperature_max` fields), we can create new one, something like `ConfigModuleUpdate`.

Reply:
I think this endpoint can replace the api/sethotwatertemperature. 
Or maybe the sethotwatertemperature support dhw_always_on also? (I dont know - lack of API docs).
I think that the idea with ConfigModuleUpdate is better. 
But It is up to You as it is your project :) And it is obvious that You are more skilled programmer than me (as I have only generic skills)

Reply:
> Or maybe the sethotwatertemperature support dhw_always_on also? (I dont know - lack of API docs).
I think that the idea with ConfigModuleUpdate is better.

I just tried it out and it didn't work - `api/sethotwatertemperature` can't change `dhw_always_on`. I suggest we move forward with the `ConfigModuleUpdate` idea then.

That is most accurate representation of the API.